azure/services/async/async_test.go

```diff
@@ -260,7 +260,7 @@ func TestCreateOrUpdateResource(t *testing.T) {
 				r.ResourceGroupName().Return("test-group")
 				s.GetLongRunningOperationState("test-resource", "test-service", infrav1.PutFuture).Return(nil)
 				c.Get(gomockinternal.AContext(), gomock.AssignableToTypeOf(&mock_azure.MockResourceSpecGetter{})).Return(nil, fakeNotFoundError)
-				r.Parameters(nil).Return(&fakeResourceParameters, nil)
+				r.Parameters(context.TODO(), nil).Return(&fakeResourceParameters, nil)
```

Comment:
for this (and all other unit test expect funcs) you can use `gomockinternal.AContext()` instead if `context.TODO()` so it works with any context that actually gets passed in, see the line above for the Get()

Reply:
Thanks! I was going to test with context.Background(). But makes sense on using `gomockinternal.AContext()` too!

Reply:
Addressed via ed9d6bba4c5dd98482b3e21a5a9f4cc2b23dfcf5

Reply:
Waiting for tests..